src/librustc/ty/layout.rs

```diff
@@ -1442,7 +1442,31 @@ impl<'tcx> LayoutCx<'tcx, TyCtxt<'tcx>> {
         debug!("generator layout ({:?}): {:#?}", ty, layout);
         Ok(layout)
     }
+}
 
+fn recompute_memory_index(offsets: &[Size], fields: &[TyLayout<'_>]) -> Vec<u32> {
```

Comment:
I wouldn't have moved this out unless it was used for anything other than the generator code.

Reply:
oh yeah it was originally being turned into a generic fn, I'll put it back

Reply:
(you can have generic `fn`s inside other functions *anyway*, FWIW)